gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionCompletionContributor.java

```diff
@@ -47,6 +52,8 @@ public class VersionCompletionContributor extends CompletionContributor {
             GroupPartOrPackageNameExplorer.getInstance();
     private final VersionExplorer versionExplorer = VersionExplorer.getInstance();
 
+    private final Queue<DependencyInfo> loadedDependencies = EvictingQueue.create(100);
```

Comment:
Why do we need this extra state? It looks like it's used just for deciding whether to add the `Loading versions...` or not. Extra state is in general bad.

I think we can just work out whether to add loading versions/no results based on the info we get from `VersionResults`. Assuming we change the signature of `VersionResults` to:

```java
public record VersionResults(Set<DependencyVersion> alreadyLoaded, Optional<CompletableFuture<Set<DependencyVersion>>> stillLoading) {}
```

 If in the combination of `VersionResults`:
* There is at least 1 `stillLoading`: add `Loading versions...`
* There are 0 `alreadyLoaded` and 0 `stillLoading`: add `No versions`
* Otherwise there are some `alreadyLoaded` and 0 `stillLoading`: add nothing, completion is complete.

Reply:
See update about the return type here: https://github.com/palantir/gradle-consistent-versions-idea-plugin/pull/80/files#r1842375371

Reply:
So I'm not sure if I'm thinking about this wrong but don't we need some state as we want to add `Loading versions` right at the top before any work is done so it pops up quickly. So we need a way at the start to determine if we should add `Loading Versions` - I guess its not essential but I think you would end up with a situation where the output said something like 

`Loading versions...`
`No versions found`